samples/doc-snippets/src/main/java/com/example/application/SessionMemoryConsumer.java

```diff
@@ -58,8 +58,8 @@ public Effect onSessionMemoryEvent(SessionMemoryEntity.Event event) {
             .method(SessionMemoryEntity::compactHistory) // <4>
             .invoke(
               new SessionMemoryEntity.CompactionCmd(
-                new SessionMessage.UserMessage(now, summary.userMessage(), ""),
-                new SessionMessage.AiMessage(now, summary.aiMessage(), ""),
+                new SessionMessage.UserMessage(now, summary.userMessage(), "", Optional.empty()), //TODO add metadata
```

Comment:
That's a big question, how do we want to present compacted history?

Reply:
I don't think we should include the metadata at all in the compaction events. Compaction events should probably be marked as "compaction" if we don't do that already. The console should probably not show the compaction interactions, or show them in a different way.

Compaction is for the whole session, not for an individual agent (componentId).

Trying to sum token counts and such is just a mess and we should not do that, especially since we allow concurrent addInteraction and compaction.